hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -139,19 +146,20 @@ private void handlePlatformTransaction(
         }
 
         // Get the consensus timestamp
-        final Instant consensusNow = platformTxn.getConsensusTimestamp();
+        final var consensusNow = platformTxn.getConsensusTimestamp();
 
         // Setup record builder list
         blockRecordManager.startUserTransaction(consensusNow, state);
         final var recordBuilder = new SingleTransactionRecordBuilder(consensusNow);
         final var recordListBuilder = new RecordListBuilder(recordBuilder);
 
+        PreHandleResult preHandleResult = null;
```

Comment:
I think the code block below has to be structured differently. It is hard to follow which block is executed under which circumstances. I am actually pretty sure that the current implementation is wrong. (For instance, if `nodeCharge` is true, we want to skip most of the rest.) But it is hard to tell.

Reply:
I think before we do a significant refactoring of the workflow, we should update the diagram and review that.